src/libsystemd/sd-daemon/sd-daemon.c

```diff
@@ -563,21 +568,27 @@ static int pid_notify_with_fds_internal(
                 }
         }
 
-        /* First try with fake ucred data, as requested */
-        if (sendmsg(fd, &msghdr, MSG_NOSIGNAL) >= 0)
-                return 1;
-
-        /* If that failed, try with our own ucred instead */
-        if (send_ucred) {
-                msghdr.msg_controllen -= CMSG_SPACE(sizeof(struct ucred));
-                if (msghdr.msg_controllen == 0)
+        while (!IOVEC_INCREMENT(msghdr.msg_iov, msghdr.msg_iovlen, n)) {
```

Comment:
hmm, this should be a do {} while() loop, no?

i.e. initially `n` is zero, hence IOVEC_INCREMENT is a nop...